### PR TITLE
fix(#9552): handle rules-engine stale state after upgrade

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

#9552 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9552-missing-aggregate/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9552-missing-aggregate/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9552-missing-aggregate/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/default/targets/target-accuracy.wdio-spec.js [hunk 1]
```diff
@@ -170,4 +172,23 @@ describe('Target accuracy', () => {
...
+        targetState: localRulesStateStore.rulesStateStore.targetState.targets
+      }
+    });
+
+    await commonPage.refresh();
+    await analyticsPage.goToTargets();
+    await commonPage.waitForPageLoaded();
+
+    const targets = await analyticsPage.getTargets();
+    console.log(targets);
```
Comment (garethbowen) [inline tests/e2e/default/targets/target-accuracy.wdio-spec.js:191]: Remove this?

Comment (dianabarsan) [reply tests/e2e/default/targets/target-accuracy.wdio-spec.js:191]: yea, I've added an additional check to this test :) it's pending the push. I'm doing one last manual upgrade test locally now, and then requesting a review. 
I am happy you're around :)

Comment (garethbowen) [reply tests/e2e/default/targets/target-accuracy.wdio-spec.js:191]: I'm always watching...

File: tests/e2e/default/targets/target-accuracy.wdio-spec.js [hunk 2]
```diff
@@ -170,4 +172,27 @@ describe('Target accuracy', () => {
...
+    await commonPage.waitForPageLoaded();
+
+    const targets = await analyticsPage.getTargets();
+    expect(targets.length).to.equal(1);
+    expect(targets[0]).to.deep.include({
+      title: 'targets.new_persons.title',
+      goal: '0',
+      count: '20',
+    });
+  });
```
Comment (garethbowen) [inline tests/e2e/default/targets/target-accuracy.wdio-spec.js:197]: Amazing diligence. Thank you.

**Review by garethbowen — APPROVED** [review]

**Pull request merged**